File: Projects/usaco/gift1.java
/*
ID: rombin82
LANG: JAVA
TASK: gift1
*/

import java.io.*;
import java.util.*;

public class gift1 {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("gift1.in"));
		PrintWriter stdout = new PrintWriter(new File("gift1.out"));
		Runtime.getRuntime().addShutdownHook(new Thread(() -> stdout.flush()));
		int name_cnt = in.nextInt();
		List<String> names = new ArrayList<String>(name_cnt);
		Map<String, Integer> table = new HashMap<String, Integer>(name_cnt);
		for (int i = 0; i < name_cnt; i++) {
			String name = in.next();
			names.add(name);
			table.put(name, 0);
		}
		for (int i = 0; i < name_cnt; i++) {
			String gvr_name = in.next();
			int money = in.nextInt();
			int rcv_cnt = in.nextInt();
			if (rcv_cnt == 0)
				continue;
			table.put(gvr_name, table.get(gvr_name) - money / rcv_cnt * rcv_cnt);
			for (int j = 0; j < rcv_cnt; j++) {
				String rcv_name = in.next();
				table.put(rcv_name, table.get(rcv_name) + money / rcv_cnt);
			}
		}
		for (String name : names)
			stdout.println(name + " " + table.get(name));
	}
}

File: Projects/usaco/gift1.kt
/*
ID: rombin82
LANG: JAVA
TASK: gift1
*/

import java.io.*
import java.util.*

object gift1 {
    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val stdin = Scanner(File("gift1.in"))
        val stdout = PrintWriter(File("gift1.out"))
        Runtime.getRuntime().addShutdownHook(Thread { stdout.flush() })
        val name_cnt = stdin.nextInt()
        val names = ArrayList<String>(name_cnt)
        val table = HashMap<String, Integer>(name_cnt)
        for (i in 0 until name_cnt) {
            val name = stdin.next()
            names.add(name)
            table.put(name, 0)
        }
        for (i in 0 until name_cnt) {
            val gvr_name = stdin.next()
            val money = stdin.nextInt()
            val rcv_cnt = stdin.nextInt()
            if (rcv_cnt == 0)
                continue
            table.put(gvr_name, table.get(gvr_name) - money / rcv_cnt * rcv_cnt)
            for (j in 0 until rcv_cnt) {
                val rcv_name = stdin.next()
                table.put(rcv_name, table.get(rcv_name) + money / rcv_cnt)
            }
        }
        for (name in names)
            stdout.println(name + " " + table.get(name))
    }
}
